### PR TITLE
rocksdb: build both regular and lite versions

### PR DESCRIPTION
**Pull request by tmm1**
cc https://github.com/Homebrew/homebrew-core/issues/9621 @ilovezfs

### CODE REVIEW

File: Formula/rocksdb.rb [hunk 1]
```diff
@@ -22,15 +22,21 @@ class Rocksdb < Formula
   def install
     ENV.cxx11
     ENV["PORTABLE"] = "1" if build.bottle?
-    if build.with? "lite"
-      ENV.append_to_cflags "-DROCKSDB_LITE=1"
-      ENV["LIBNAME"] = "librocksdb_lite"
-    end
-    system "make", "clean"
```
Comment (ilovezfs) [inline Formula/rocksdb.rb]: do we still need to run make clean here too?

File: Formula/rocksdb.rb [hunk 2]
```diff
@@ -11,7 +12,6 @@ class Rocksdb < Formula
     sha256 "b058bea89af8825ab24bcebe61418ed20a8d504805292dda9d5addca181d13e1" => :yosemite
   end
 
-  option "without-lite", "Don't build mobile/non-flash optimized lite version"
   option "with-tools", "Build tools"
```
Comment (ilovezfs) [inline Formula/rocksdb.rb]: any thoughts on always installing the tools too?

File: Formula/rocksdb.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@ class Rocksdb < Formula
   homepage "http://rocksdb.org"
   url "https://github.com/facebook/rocksdb/archive/v5.0.2.tar.gz"
   sha256 "5e39d2131ebdb92c30eda3d32861f489f4843fad50cc2fbd5d234bc4415948ca"
+  revision 2
```
Comment (ilovezfs) [inline Formula/rocksdb.rb]: revision 1 (we count from 0 implicitly)

File: Formula/rocksdb.rb (hunk 2 above)
Comment (tmm1) [inline Formula/rocksdb.rb]: Yea that seems like a good idea. Should I keep the option (`without-tools`) or remove it?

File: Formula/rocksdb.rb (hunk 1 above)
Comment (tmm1) [inline Formula/rocksdb.rb]: Are the build directories ever re-used? On a fresh build `make clean` shouldn't be necessary.

File: Formula/rocksdb.rb [hunk 4]
```diff
@@ -22,15 +22,21 @@ class Rocksdb < Formula
...
-    if build.with? "lite"
-      ENV.append_to_cflags "-DROCKSDB_LITE=1"
-      ENV["LIBNAME"] = "librocksdb_lite"
-    end
-    system "make", "clean"
+
+    # build regular rocksdb
     system "make", "static_lib"
     system "make", "shared_lib"
     system "make", "tools" if build.with? "tools"
```
Comment (ilovezfs) [inline Formula/rocksdb.rb]: are the tools identical for lite and non-lite?

File: Formula/rocksdb.rb (hunk 1 above)
Comment (ilovezfs) [inline Formula/rocksdb.rb]: nope, but I have no idea why that make clean was originally there

File: Formula/rocksdb.rb (hunk 2 above)
Comment (ilovezfs) [inline Formula/rocksdb.rb]: You'd just remove it. We only use "deprecated_option" when it's being mapped to something other than nothing.

**Comment by tmm1** [comment]
> are the tools identical for lite and non-lite?

I believe the regular tools would work with both, since they contain a superset of lite features.

@igorcanadi @theopolis can you confirm?

File: Formula/rocksdb.rb (hunk 1 above)
Comment (tmm1) [inline Formula/rocksdb.rb]: Ok I put it back.

File: Formula/rocksdb.rb (hunk 2 above)
Comment (tmm1) [inline Formula/rocksdb.rb]: Alright, removed.

File: Formula/rocksdb.rb (hunk 3 above)
Comment (tmm1) [inline Formula/rocksdb.rb]: thanks

File: Formula/rocksdb.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/rocksdb.rb]: > Are the build directories ever re-used?

They are not.

**Comment by tmm1** [comment]
Looks like the tests for the tools are failing- doesn't seem like they ever worked, but that was masked by the fact that the tools weren't built by default.

**Comment by tmm1** [comment]
> I believe the regular tools would work with both, since they contain a superset of lite features.

Actually it looks like most of the tools do not even support LITE mode:

https://github.com/facebook/rocksdb/blob/master/tools/dump/rocksdb_dump.cc#L14
https://github.com/facebook/rocksdb/blob/master/tools/dump/rocksdb_undump.cc#L14
https://github.com/facebook/rocksdb/blob/master/tools/ldb.cc#L18
https://github.com/facebook/rocksdb/blob/master/tools/sst_dump.cc#L18

**Comment by tmm1** [comment]
@ilovezfs Can this be merged? Happy to address any remaining concerns.

**Comment by ilovezfs** [comment]
@tmm1 do you know why the tools tests fail? Is it just that the exit code isn't zero for `--help`? If so, you can use `assert_match "something expected", shell_output("#{bin}/foo --help", 1)` assuming the exit code were 1 not 0.

**Comment by tmm1** [comment]
Yes they all exit 1. I can switch to assertions, thanks for the tip.

**Comment by ilovezfs** [comment]
Great, thanks!

**Comment by tmm1** [comment]
@ilovezfs All set.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Shipped! Thanks @tmm1. This formula is much better now.

**Comment by theopolis** [comment]
Thanks for the work here folks! Excited that Rocks is now more flexible

**Pull request closed without merging**